Uplift/Views/GymDetail/GymDetailTimeInfoView.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  GymDetailTimeInfoView.swift
+//  Uplift
+//
+//  Created by Phillip OReggio on 10/22/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+
+class GymDetailTimeInfoView: UILabel {
+
+    private var facility: Facility
+    private var selectedDayIndex = 0
+    private var displayedText = ""
+    private var timesText = NSMutableAttributedString()
+    private var paragraphStyle = NSMutableParagraphStyle()
+
+    // MARK: - Init
+    init(facility: Facility) {
+        self.facility = facility
+        super.init(frame: CGRect())
+
+        timesText.mutableString.setString(displayedText)
+
+        paragraphStyle.lineSpacing = 5.0
+        paragraphStyle.maximumLineHeight = 26
+        paragraphStyle.alignment = .center
+
+        attributedText = timesText
+        backgroundColor = .primaryWhite
+        numberOfLines = 0
+        font = ._16MontserratRegular
+        textColor = .primaryBlack
+        textAlignment = .center
+
+        updateAppearance()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Update
+    private func updateAppearance() {
+        // func call to update tags
+        updateTimes()
+    }
+
+    private func updateTimes() {
+        displayedText = makeDisplayText(dayIndex: selectedDayIndex)
+        updateAttributedText()
+    }
+
+    private func updateTags() {
+        let tagLabelWidth = 81
+        let tagSideOffset = 25.0
+        let textLineHeight = font.lineHeight
+
+        let info = facility.miscInformation
+        subviews.forEach({ $0.removeFromSuperview() })
+
+        for i in 0..<info.count {
+            if info[i] == "" { // Don't use blank tags
+                continue
+            } else {
+                let infoView = AdditionalInfoView()
+                let spacing = textLineHeight * CGFloat(i)
+                let inset: CGFloat = 2
+
+                infoView.text = info[i]
+                addSubview(infoView)
+                infoView.snp.makeConstraints { make in
+                    make.top.equalToSuperview().inset(spacing + inset)
+                    make.trailing.equalToSuperview().inset(tagSideOffset)
+                    make.width.equalTo(tagLabelWidth)
+                }
```

Comment:
if we're adding subviews here, we should make this a subclass of `UIView` instead